### PR TITLE
Alex branch

### PR DESCRIPTION
**Pull request by AlexSyt**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/views/MapControlsLayer.java [hunk 1]
```diff
@@ -350,6 +350,10 @@ protected void clickRouteWaypoints() {
 		}
 	}
 
+	public void stopNavigation() {
+		clickRouteCancel();
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/views/MapControlsLayer.java]: Extract clickRouteCancel() to stopNavigation() and call stopNavigation() from clickRouteCancel()

File: OsmAnd/src/net/osmand/plus/views/MapControlsLayer.java [hunk 2]
```diff
@@ -440,6 +444,10 @@ public void doNavigate() {
 		startNavigation();
 	}
 
+	public void showNavigationDialog() {
+		onNavigationClick();
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/views/MapControlsLayer.java]: Why you do not call doRoute(boolean hasTargets) instead?

File: OsmAnd/src/net/osmand/plus/quickaction/QuickActionFactory.java [hunk 3]
```diff
@@ -46,7 +46,7 @@
 
     public String quickActionListToString(List<QuickAction> quickActions) {
 
-		return new Gson().toJson(quickActions);
+        return new Gson().toJson(quickActions);
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/quickaction/QuickActionFactory.java]: Fixed with spaces?

Comment (AlexSyt) [inline OsmAnd/src/net/osmand/plus/quickaction/QuickActionFactory.java]: yes

File: OsmAnd/src/net/osmand/plus/views/MapControlsLayer.java (hunk 2 above)
Comment (AlexSyt) [inline OsmAnd/src/net/osmand/plus/views/MapControlsLayer.java]: Dima said that the behavior when clicking on the navigation button and navigation in quick actions should be identical. Therefore I use the same method.

File: OsmAnd/src/net/osmand/plus/views/MapControlsLayer.java (hunk 1 above)
Comment (AlexSyt) [inline OsmAnd/src/net/osmand/plus/views/MapControlsLayer.java]: Fixed.

**Pull request closed without merging**